metrics/providers/micrometer/src/main/java/io/helidon/metrics/providers/micrometer/MMeterRegistry.java

```diff
@@ -343,16 +379,21 @@ io.micrometer.core.instrument.MeterRegistry delegate() {
 
     @Override
     public Iterable<io.helidon.metrics.api.Meter> meters(Iterable<String> scopeSelection) {
-        if (scopeSelection.iterator().hasNext()) {
-            Set<io.helidon.metrics.api.Meter> result = new HashSet<>();
-            for (String scope : scopeSelection) {
-                if (scopeMembership.containsKey(scope)) {
-                    result.addAll(scopeMembership.get(scope));
+        lock.readLock().lock();
+        try {
+            if (scopeSelection.iterator().hasNext()) {
+                Set<io.helidon.metrics.api.Meter> result = new HashSet<>();
+                for (String scope : scopeSelection) {
+                    if (scopeMembership.containsKey(scope)) {
+                        result.addAll(scopeMembership.get(scope));
+                    }
                 }
+                return result;
```

Comment:
Watch mutability.

Reply:
As above, we create a new result set that is not backed by our internal set, so I'm not sure why we'd restrict what the caller can do with the return value.